src/mvAppItem.cpp

```diff
@@ -2660,32 +2699,35 @@ DearPyGui::GetEntityParser(mvAppItemType type)
         args.push_back({ mvPyDataType::Bool, "no_box_select", mvArgType::KEYWORD_ARG, "False", "the user will not be able to box-select with right-click drag"});
-        args.push_back({ mvPyDataType::Bool, "no_mouse_pos", mvArgType::KEYWORD_ARG, "False", "the mouse position, in plot coordinates, will not be displayed inside of the plot"});
+        args.push_back({ mvPyDataType::Bool, "no_mouse_text", mvArgType::KEYWORD_ARG, "False", "the text of mouse position, in plot coordinates, will not be displayed inside of the plot"});
         args.push_back({ mvPyDataType::Bool, "no_highlight", mvArgType::KEYWORD_ARG, "False", "plot items will not be highlighted when their legend entry is hovered"});
-        args.push_back({ mvPyDataType::Bool, "no_child", mvArgType::KEYWORD_ARG, "False", "a child window region will not be used to capture mouse scroll (can boost performance for single ImGui window applications)"});
         args.push_back({ mvPyDataType::Bool, "query", mvArgType::KEYWORD_ARG, "False", "the user will be able to draw query rects with middle - mouse or CTRL + right - click drag"});
         args.push_back({ mvPyDataType::Bool, "crosshairs", mvArgType::KEYWORD_ARG, "False", "the default mouse cursor will be replaced with a crosshair when hovered"});
-        args.push_back({ mvPyDataType::Bool, "anti_aliased", mvArgType::KEYWORD_ARG, "False", "plot lines will be software anti-aliased (not recommended for high density plots, prefer MSAA)"});
         args.push_back({ mvPyDataType::Bool, "equal_aspects", mvArgType::KEYWORD_ARG, "False", "primary x and y axes will be constrained to have the same units/pixel (does not apply to auxiliary y-axes)"});
+        args.push_back({ mvPyDataType::Bool, "no_legend", mvArgType::KEYWORD_ARG, "False", "the legend will not be displayed"});
+        args.push_back({ mvPyDataType::Bool, "no_inputs", mvArgType::KEYWORD_ARG, "False", "the user will not be able to interact with the plot"});
+        args.push_back({ mvPyDataType::Bool, "no_frame", mvArgType::KEYWORD_ARG, "False", "the ImGui frame will not be rendered"});
+        // args.push_back({ mvPyDataType::Bool, "canvas_only", mvArgType::KEYWORD_ARG, "False", "shortcut for: no_title + no_legend + no_menu + no_box_select + no_mouse_text"});
         args.push_back({ mvPyDataType::Bool, "use_local_time", mvArgType::KEYWORD_ARG, "False", "axis labels will be formatted for your timezone when" });
         args.push_back({ mvPyDataType::Bool, "use_ISO8601", mvArgType::KEYWORD_ARG, "False", "dates will be formatted according to ISO 8601 where applicable (e.g. YYYY-MM-DD, YYYY-MM, --MM-DD, etc.)" });
         args.push_back({ mvPyDataType::Bool, "use_24hour_clock", mvArgType::KEYWORD_ARG, "False", "times will be formatted using a 24 hour clock" });
+        args.push_back({ mvPyDataType::Bool, "delete_rect", mvArgType::KEYWORD_ARG, "True", "allows to delete drag rect with double left mouse click" });
```

Comment:
`delete_rect` - a number of concerns with this argument:

- :exclamation::exclamation: `delete_rect` is a rather confusing name. Like, I don't know what to expect when I see `dpg.plot(delete_rect=False)`.
- Thinking about the purpose/use case for this flag: right now, if we specify `delete_rect=False`, this means the user will not be able to delete query rects, but will still be able to create them. Suppose a plot has a query rect from the very start, and the user by mistake creates one more query rect. If deletion is disabled, how would the user cancel the mistaken action?
\
My feeling is that to properly customize this behavior, the option should disable _both_ "add" and "delete" actions. That is, either the user can freely add _and_ delete rectangles, or they can't do either of those (but will still be able to drag the rectangles).

  - Thinking further: if the user can add/delete rects, it might be useful to restrict the number of rects **both** from above and from below.
\
In the end, it looks like a better way to do it might be to mimic the node editor and how it adds links via a separate callback. Like, instead of adding a rect to an internal list, we just call the user's callback and then the application can add a `drag_rect` on its own.
\
The downside is that it would be totally incompatible with the old query feature. We can probably support both?
- With the default button assignment (`SelectCancel` is the left button), the double-click behavior of `delete_rect` conflicts with double-clicks on DragRect: you can double-click the edge to make it snap to the view boundary. With `delete_rect=True`, it looks really weird: the rect first expands and then disappears. Also, that edge expansion is a nice feature that we probably don't want to lose.
- Double-click deletes the last rect no matter what item was clicked, whereas intuitively the user would expect that they double-click a rect and _exactly that rect_ gets deleted.

In the end, this argument, as well as the entire query/drag_rect change, needs more work to make it more consistent.

Reply:
Okay so, I reworked a little bit this thing
First: I removed delete rect giving for granted that if you can create them you can also delete them (as you said)
Second: now with the double click you actually delete only the rect you're hovering

> Thinking further: if the user can add/delete rects, it might be useful to restrict the number of rects both from above and from below.
>In the end, it looks like a better way to do it might be to mimic the node editor and how it adds links via a separate callback. Like, instead of adding a rect to an internal list, we just call the user's callback and then the application can add a drag_rect on its own.

I'm not really into the node editor so I'd prefer to leave it simple like this and then if you or someone wants to change it, perfect

> With the default button assignment (SelectCancel is the left button), the double-click behavior of delete_rect conflicts with double-clicks on DragRect: you can double-click the edge to make it snap to the view boundary. With delete_rect=True, it looks really weird: the rect first expands and then disappears. Also, that edge expansion is a nice feature that we probably don't want to lose.

I actually didn't notice that


The only problem left is that the double clicking also triggers the auto fitting of the plot, I don't know how this can be avoided

Reply:
Here's an update that fixes the issue with undesirable zooming on the double-click, and also makes double-click work on the middle point of the drag rect, too.

https://github.com/v-ein/DearPyGui-fixes/compare/8793b0c932b98827307797919c3a427c9808e24e...feature/plot-query-rect

Reply:
That's a really nice update, thanks! I'll push it now

Reply:
I've just pushed one more commit to this branch, fixing a couple of issues with how the callback is called:

1. It was not called when a rect is added or deleted (i.e. was only called when a rect is moved around).
2. For some reason there was an extra loop causing the callback to be called with the same `app_data` multiple times.

No more changes are expected, feel free to pick up that branch now.

Reply:
@SamuMazzi would you please pick up that [commit ac789ad](https://github.com/v-ein/DearPyGui-fixes/commit/ac789ade616903d7c1bf12c2a2a2b0aae1e094c4) I mentioned in my previous comment?

Reply:
I lost this comment too, just did it, thanks!